### PR TITLE
Solace Finance adapter refactoring

### PR DESCRIPTION
**Pull request by ssozuer**
##### List of audit links if any:
https://solace.fi/files/Solace-Quantstamp-Report.pdf

https://hacken.io/audits/#solace

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
https://drive.google.com/file/d/1gdQzwwoq4duE1qPsRQHYzUubxlewhIkJ/view?usp=sharing
https://drive.google.com/file/d/1Whew8s2lMDitr7Zd_VnGCGyRk8msuILg/view?usp=sharing

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list) 
solace

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
15735

##### Short Description (to be shown on DefiLlama):
Solace protects your funds against smart-contracts exploits across 180+ protocols with an intelligent single policy that automatically adjusts coverage to changes in your portfolio. Solace underwrites risks with its owned capital, so the revenue from protocol's activities goes to stakers.

##### Token address and ticker if any:
0x501acE9c35E60f03A2af4d484f49F9B1EFde9f40 $SOLACE

##### methodology (what is being counted as tvl, how is tvl being calculated):
Assets locked in the underwriting pool and staked $SOLACE

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/solace-finance exports TVL: 
        
 
 
    polygon                   22.49 k
    aurora                    409.50 k
    ethereum                  864.08 k
    
    total                    1.30 M

File: projects/solace-finance/index.js [hunk 1]
```diff
@@ -35,36 +39,52 @@ const ETHEREUM_TOKENS = [
...
+    
+    sdk.util.sumSingleBalance(balances, address, balance);
   });
   return balances;
 }
 
+async function ethereum(timestamp, block, chainBlocks) {
+  let result = await fetchBalances(ETHEREUM_UWP_ADDRESS, ETHEREUM_TOKENS,  "ethereum", chainBlocks["ethereum"]);
+  const lpBalances = await fetchBalances(SOLACE_USDC_POOL, ETHEREUM_TOKENS,  "ethereum", chainBlocks["ethereum"]);
+  result[ETHEREUM_LP_TOKENS.SOLACE] = BigNumber(result[ETHEREUM_LP_TOKENS.SOLACE] || 0).plus(lpBalances[ETHEREUM_LP_TOKENS.SOLACE] || 0).toFixed();
```
Comment (waynebruce0x) [inline projects/solace-finance/index.js:76]: do you need to unwrap these LP tokens using unwrapUniswapLPs()?

**Comment by llamatester** [comment]
The adapter at projects/solace-finance exports TVL: 
        
 
 
    polygon                   213.10 k
    ethereum                  3.96 M
    aurora                    844.42 k
    
    total                    5.02 M

File: projects/solace-finance/index.js (hunk 1 above)
Comment (ssozuer) [reply projects/solace-finance/index.js:80]: No we don't need actually. I tried it but it didn't work. We're on multiple chains and using different lp structures.

**Comment by waynebruce0x** [comment]
can the LP be exported separately as 'pool2'?

**Comment by ssozuer** [comment]
> can the LP be exported separately as 'pool2'?

In the previous version, at least I tried it with ethereum. The polygon and aurora didn't work also it broke our tvl value in coingecko. [The coingecko was only fetching ](https://www.coingecko.com/en/coins/solace).

**Comment by g1nt0ki** [comment]
@ssozuer sorry, but I am closing this in favor of https://github.com/DefiLlama/DefiLlama-Adapters/pull/2643

Polygon lp was not added because it was uniswap v3 version, other LPs are parsed properly in added to pool2 in the other PR.

Question: is the polygon LP added by the users?

**Comment by g1nt0ki** [comment]
I am asking because I dont see any uni v3 nfts on this address 0xd1108a800363c262774b990e9df75a4287d5c075

**Comment by ssozuer** [comment]
> 

0xd1108a800363C262774B990e9DF75a4287d5c075 is our polygon underwriting pool address. Our LP address is 0x85Efec4ee18a06CE1685abF93e434751C3cb9bA9 in polygon and it is created by solace protocol.

**Pull request closed without merging**